python/python/tests/test_pubsub.py

```diff
@@ -148,6 +156,64 @@ def new_message(msg: CoreCommands.PubSubMsg, context: Any):
     received_messages.append(msg)
 
 
+async def client_cleanup(
+    client: Union[GlideClient, GlideClusterClient],
+    cluster_mode_subs: Optional[
+        GlideClusterClientConfiguration.PubSubSubscriptions
+    ] = None,
+):
+    """
+    This function tries its best to clear state assosiated with client
+    Its explicitly calls client.close() and deletes the object
+    In addition, it tries to clean up cluster mode subsciptions since it was found the closing the client via close() is not enough.
+    Note that unsubscribing is not feasible in the current implementation since its unknown on which node the subs are configured
+    """
+    # if is_cluster:
+    #     pubsub_subs = cast(ClusterClientConfiguration.PubSubSubscriptions, pubsub_subs)
+    # else :
+    #     pubsub_subs = cast(GlideClientConfiguration.PubSubSubscriptions, pubsub_subs)
+
+    # for channel_type, channel_patterns in pubsub_subs:
+    #     if channel_type == ClusterClientConfiguration.PubSubChannelModes.Exact or channel_type == GlideClientConfiguration.PubSubChannelModes.Exact:
+    #         cmd = "UNSUBSCRIBE"
+    #     elif channel_type == ClusterClientConfiguration.PubSubChannelModes.Pattern or channel_type == GlideClientConfiguration.PubSubChannelModes.Pattern:
+    #         cmd = "PUNSUBSCRIBE"
+    #     else:
+    #         cmd = "SUNSUBSCRIBE"
+
+    #     # we need unsubscribe commands because close might
+    #     # UNSUBSCRIBE commands are unsupported, also, the routing might be wrong in cluster mode
+    #     for channel_patern in channel_patterns:
+
+    #         await listening_client.custom_command(
+    #             ["UNSUBSCRIBE", *list(channels_and_messages.keys())]
+    #         )
+
+    if cluster_mode_subs:
+        for (
+            channel_type,
+            channel_patterns,
+        ) in cluster_mode_subs.channels_and_patterns.items():
+            if channel_type == GlideClusterClientConfiguration.PubSubChannelModes.Exact:
+                cmd = "UNSUBSCRIBE"
+            elif (
+                channel_type
+                == GlideClusterClientConfiguration.PubSubChannelModes.Pattern
+            ):
+                cmd = "PUNSUBSCRIBE"
+            elif not await check_if_server_version_lt(client, "7.0.0"):
+                cmd = "SUNSUBSCRIBE"
+            else:
+                continue
```

Comment:
when can we get to this branch? why do we ignore and just continue?

Reply:
please make it more readable by matching the sharded option. if new pubsub modes will be added it will get hard to understand failures due to this if/else together with the engine version etc

Reply:
we need it because client_up() may be called with shared configuration which is used only with server version 7.0+

Reply:
added comment